src/cogent3/core/location.py

```diff
@@ -1077,6 +1079,101 @@ def _input_vals_neg_step(
     return (0, 0, 1) if start < stop else (start, stop, step)
 
 
+@numba.jit
+def seq_to_align_index(
+    gap_pos: IntArrayTypes,
+    cum_lengths: IntArrayTypes,
+    parent_length: int,
+    num_gaps: int,
+    seq_index: int,
+    slice_stop: bool = False,
+) -> int:
+    """convert a sequence index into an alignment index
+
+    Parameters
+    ----------
+    seq_index
+        coordinate on the sequence, must be < parent_length
+    slice_stop
+        set to True if the index is to be the end of an alignment slice.
+        In that case, and if seq_index is in gap_pos then it returns
+        the first alignment index of the gap run.
+    """
+    # NOTE I explicitly cast all returned values to python int's due to
+    # need for json serialisation, which does not support numpy int classes
+    if seq_index < 0:
+        seq_index += parent_length
+
+    if seq_index < 0:
+        raise IndexError(f"{seq_index} negative seq_index beyond limit ")
+
+    if not num_gaps or seq_index < gap_pos[0]:
+        return int(seq_index)
+
+    # if stop_index, check if the seq_index corresponds to a gap position
```

Comment:
very minor, but what is the `stop_index` referred to in this comment?

Reply:
it should be slice_stop, good catch!